<a href="https://colab.research.google.com/github/B-Murali-Krishna/Hotel-Booking-Analysis---Capstone-Project/blob/main/Hotel_Booking_Analysis_Capstone_Project_Final.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## <b> Have you ever wondered when the best time of year to book a hotel room is? Or the optimal length of stay in order to get the best daily rate? What if you wanted to predict whether or not a hotel was likely to receive a disproportionately high number of special requests? This hotel booking dataset can help you explore those questions!

## <b>This data set contains booking information for a city hotel and a resort hotel, and includes information such as when the booking was made, length of stay, the number of adults, children, and/or babies, and the number of available parking spaces, among other things. All personally identifying information has been removed from the data. </b>

## <b> Explore and analyze the data to discover important factors that govern the bookings. </b>


Mount the Google Drive to Google Colab Notebook.


In [1]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


Import the main libraries that we will use for this project.


In [2]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

# **Data Understanding**

Import the dataset into colab notebook and display top 5 rows.

In [4]:
df=pd.read_csv('/content/drive/MyDrive/Alma Better/AlmaBetter CapstoneProject/EDA on Hotel Booking/Hotel Bookings.csv')
df.head()

hotel  is_canceled  ...  reservation_status  reservation_status_date
0  Resort Hotel            0  ...           Check-Out               2015-07-01
1  Resort Hotel            0  ...           Check-Out               2015-07-01
2  Resort Hotel            0  ...           Check-Out               2015-07-02
3  Resort Hotel            0  ...           Check-Out               2015-07-02
4  Resort Hotel            0  ...           Check-Out               2015-07-03

[5 rows x 32 columns]


Display the last 5 rows.

In [5]:
df.tail()

hotel  is_canceled  ...  reservation_status  reservation_status_date
119385  City Hotel            0  ...           Check-Out               2017-09-06
119386  City Hotel            0  ...           Check-Out               2017-09-07
119387  City Hotel            0  ...           Check-Out               2017-09-07
119388  City Hotel            0  ...           Check-Out               2017-09-07
119389  City Hotel            0  ...           Check-Out               2017-09-07

[5 rows x 32 columns]

Check the number of rows and columns present in our dataset.

In [6]:
df.shape

(119390, 32)

* We have 119390 rows and 32 features in our dataset.

Let's check what all columns are present in our dataset.

In [7]:
df.columns

Index(['hotel', 'is_canceled', 'lead_time', 'arrival_date_year',
       'arrival_date_month', 'arrival_date_week_number',
       'arrival_date_day_of_month', 'stays_in_weekend_nights',
       'stays_in_week_nights', 'adults', 'children', 'babies', 'meal',
       'country', 'market_segment', 'distribution_channel',
       'is_repeated_guest', 'previous_cancellations',
       'previous_bookings_not_canceled', 'reserved_room_type',
       'assigned_room_type', 'booking_changes', 'deposit_type', 'agent',
       'company', 'days_in_waiting_list', 'customer_type', 'adr',
       'required_car_parking_spaces', 'total_of_special_requests',
       'reservation_status', 'reservation_status_date'],
      dtype='object')

# **Data Preparation and Cleaning**

Let's check if there are duplicates in our dataset.

In [8]:
len(df[df.duplicated()])

31994

* We have large number of duplicates in our dataset. So removing these duplictes would be better.

In [9]:
df= df.drop_duplicates()
len(df[df.duplicated()])

0

* Now we don't have any duplicate in our dataset.

In [10]:
df.shape

(87396, 32)